File: spotify_clone_app/lib/features/auth/dtos/login_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'login_dto.g.dart';

@JsonSerializable()
class LoginDto {
  final String email;
  final String password;
  final String deviceId;
  final String deviceInfo;

  LoginDto({
    required this.email,
    required this.password,
    required this.deviceId,
    required this.deviceInfo,
  });

  Map<String, dynamic> toJson() => _$LoginDtoToJson(this);

  factory LoginDto.fromJson(Map<String, dynamic> json) => _$LoginDtoFromJson(json);
}

File: spotify-backend/src/auth/dtos/login.dto.ts
import { IsEmail, IsNotEmpty, IsOptional, IsString, MinLength } from 'class-validator';

export class LoginDto {
    @IsEmail()
    email: string;

    @IsNotEmpty()
    @MinLength(6)
    password: string;

    @IsString()
    @IsNotEmpty()
    deviceId: string;

    @IsString()
    @IsOptional()
    deviceInfo?: string;
}
